e2e/cmd/e2e.go

```diff
@@ -184,6 +192,9 @@ func main() {
 			// not the logic of KSM. This might change if KSM sharding becomes enabled by default.
 			s.TwoKSMInstances = true
 		},
+		"with_namespace_filtering": func(s *scenario.Scenario) {
+			s.ExtraArgs = []string{"-kube_state_metrics_namespace=default"}
```

Comment:
Outside of the context I wouldn't be a fan of `Extra` stuff but I do think it's the lesser evil here. Nice job on that.

Reply:
Can you elaborate on "Extra stuff"? I'm not sure what you mean.